zkevm-circuits/src/anchor_tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,465 @@
+            let sig = meta.query_advice(sig, Rotation::cur());
+
+            cb.require_equal("sig_u64_acc[start] = sig[start]", sig_u64_acc, sig);
+            cb.gate(q_u64_start)
+        });
+
+        // Check that R has the expected value
+        meta.create_gate(
+            "IF r == GX2 THEN a(msg_hash) * b(1) + c(GX1_MUL_PRIVATEKEY) == d(N)",
+            |meta| {
+                let mut cb = BaseConstraintBuilder::new(MAX_DEGREE);
+
+                // R needs to be either GX1 or Gx2
+                let sig_rlc_acc = meta.query_advice(sig_rlc_acc, Rotation(63));
+                cb.require_in_set("r in (GX1, GX2)", sig_rlc_acc, vec![gx1_rlc, gx2_rlc]);
+
+                // In mul_add chip, we have a * b + c == d
+                // => a == msg_hash
+                // => b == 1
+                // => c == GX1_MUL_PRIVATEKEY
+                // => d == N
+                // Check that the inputs a, b, and c into mul_add are correct
+                // a == msg_hash
+                let a = mul_add.a_limbs_cur(meta);
+                cb.require_equal("a.0", a.0, meta.query_advice(sig_u64_acc, Rotation(31)));
+                cb.require_equal("a.1", a.1, meta.query_advice(sig_u64_acc, Rotation(23)));
+                cb.require_equal("a.2", a.2, meta.query_advice(sig_u64_acc, Rotation(15)));
+                cb.require_equal("a.3", a.3, meta.query_advice(sig_u64_acc, Rotation(7)));
+                // b == 1
+                let b = mul_add.b_limbs_cur(meta);
+                let one = split_u256_limb64(&U256::one())
+                    .map(|v| Expression::Constant(F::from(v.as_u64())));
+                cb.require_equal("b.0", b.0, one[0].expr());
+                cb.require_equal("b.1", b.1, one[1].expr());
+                cb.require_equal("b.2", b.2, one[2].expr());
+                cb.require_equal("b.3", b.3, one[3].expr());
+                // c == GX1_MUL_PRIVATEKEY
+                let gx1_mul_privatekey_0 =
+                    Expression::Constant(F::from_u128(GX1_MUL_PRIVATEKEY_LO_HI.0.as_u128()));
+                let gx1_mul_privatekey_1 =
+                    Expression::Constant(F::from_u128(GX1_MUL_PRIVATEKEY_LO_HI.1.as_u128()));
+                let c = mul_add.c_lo_hi_cur(meta);
+                cb.require_equal("c.0", c.0, gx1_mul_privatekey_0);
+                cb.require_equal("c.1", c.1, gx1_mul_privatekey_1);
+
+                // Now finally check that d == N
+                let n_0 = Expression::Constant(F::from_u128(N_LO_HI.0.as_u128()));
+                let n_1 = Expression::Constant(F::from_u128(N_LO_HI.1.as_u128()));
+                let d = mul_add.d_lo_hi_cur(meta);
+                cb.require_equal("d.0", d.0, n_0);
+                cb.require_equal("d.1", d.1, n_1);
+
+                cb.gate(meta.query_selector(q_check))
+            },
+        );
+
+        // Range constraint all bytes
+        meta.lookup_any("ensure all bytes are actually byte values", |meta| {
```

Comment:
Just simplified this one so the lookup is always done. This is fine because the column is always used to store just byte values, or is not used in which case it just stores `0`.